src/sentry/workflow_engine/migration_helpers/issue_alert_migration.py

```diff
@@ -0,0 +1,222 @@
+import logging
+from dataclasses import dataclass
+from typing import Any
+
+from sentry.grouping.grouptype import ErrorGroupType
+from sentry.models.rule import Rule
+from sentry.rules.conditions.event_frequency import EventUniqueUserFrequencyConditionWithConditions
+from sentry.rules.processing.processor import split_conditions_and_filters
+from sentry.workflow_engine.migration_helpers.issue_alert_conditions import (
+    create_event_unique_user_frequency_condition_with_conditions,
+    translate_to_data_condition,
+)
+from sentry.workflow_engine.migration_helpers.rule_action import (
+    build_notification_actions_from_rule_data_actions,
+)
+from sentry.workflow_engine.models import (
+    AlertRuleDetector,
+    AlertRuleWorkflow,
+    DataCondition,
+    DataConditionGroup,
+    DataConditionGroupAction,
+    Detector,
+    DetectorWorkflow,
+    Workflow,
+    WorkflowDataConditionGroup,
+)
+from sentry.workflow_engine.models.data_condition import (
+    Condition,
+    enforce_data_condition_json_schema,
+)
+
+logger = logging.getLogger(__name__)
+
+SKIPPED_CONDITIONS = [Condition.EVERY_EVENT]
+
+
+@dataclass
+class IssueAlertMigrator:
+    def __init__(self, rule: Rule, user_id: int | None = None, is_dry_run: bool | None = False):
+        self.rule = rule
+        self.user_id = user_id
+        self.is_dry_run = is_dry_run
+        self.data = rule.data
+        self.project = rule.project
+        self.organization = self.project.organization
+
+    def run(self) -> None:
+        error_detector = self._create_detector_lookup()
+        conditions, filters = split_conditions_and_filters(self.data["conditions"])
+        workflow = self._create_workflow_and_lookup(
+            conditions=conditions,
+            filters=filters,
+            action_match=self.data["action_match"],
+            detector=error_detector,
+        )
+        if_dcg = self._create_if_dcg(
+            filter_match=self.data.get("filter_match", "all"),
```

Comment:
default rules don't have filter_match
https://github.com/getsentry/sentry/blob/c62cc130acc863a96181b4fe5d7673aa2dbc7f40/src/sentry/receivers/rules.py#L20-L27

but the UI defaults this to ALL
<img width="969" alt="Screenshot 2025-02-18 at 15 24 12" src="https://github.com/user-attachments/assets/7537ac6c-602a-4f70-bf04-b1be373606f9" />